### PR TITLE
Integrate bats-assert + fix SC issues in test

### PR DESCRIPTION
**Pull request by timja**
[bats-assert](https://github.com/bats-core/bats-assert) will give us the actual error when tests fail, unlike in https://github.com/jenkinsci/docker-agent/pull/193 where the locale is broken but it just says test failed.

we already use it in https://github.com/jenkinsci/docker

### CODE REVIEW

File: tests/tests.bats [hunk 1]
```diff
@@ -1,16 +1,17 @@
 #!/usr/bin/env bats
 
 load test_helpers
+load 'test_helper/bats-support/load' # this is required by bats-assert!
+load 'test_helper/bats-assert/load'
 
-REGEX='^([0-9]+)/(.+)$'
-
+IMAGE=${IMAGE:-debian_jdk11}
```
Comment (timja) [inline tests/tests.bats:7]: supply a default value for when running outside of `make`, (like in IDE)

File: tests/tests.bats [hunk 2]
```diff
@@ -1,16 +1,17 @@
 #!/usr/bin/env bats
 
 load test_helpers
+load 'test_helper/bats-support/load' # this is required by bats-assert!
+load 'test_helper/bats-assert/load'
 
-REGEX='^([0-9]+)/(.+)$'
```
Comment (timja) [inline tests/tests.bats:5]: unused variable

**Review by MarkEWaite — APPROVED** [review]
Looks good to me, though I agree that the two comments might be nice to include before merge

**Comment by timja** [comment]
> Looks good to me, though I agree that the two comments might be nice to include before merge

2 comments?

**Comment by MarkEWaite** [comment]
> > Looks good to me, though I agree that the two comments might be nice to include before merge
> 
> 2 comments?

https://github.com/jenkinsci/docker-agent/pull/194#discussion_r702288421 and 

> > Looks good to me, though I agree that the two comments might be nice to include before merge
> 
> 2 comments?

I missed that your review comments were not proposing any change, just explaining the rationale for the change in that line

File: .gitmodules [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[submodule "tests/test_helper/bats-assert"]
+	path = tests/test_helper/bats-assert
+	url = https://github.com/bats-core/bats-assert.git
```
Comment (timja) [inline .gitmodules:3]:
```suggestion
	url = https://github.com/bats-core/bats-assert
```

File: .gitmodules [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+[submodule "tests/test_helper/bats-support"]
+	path = tests/test_helper/bats-support
+	url = https://github.com/bats-core/bats-support.git
```
Comment (timja) [inline .gitmodules:6]:
```suggestion
	url = https://github.com/bats-core/bats-support
```

**Comment by timja** [comment]
Created a new PR due to issues on CI, ref https://newbedev.com/git-clone-works-git-submodule-fails-permission-denied

**Pull request closed without merging**